boxoffice/models/invoice.py

```diff
@@ -72,6 +73,17 @@ def __init__(self, *args, **kwargs):
         self.invoice_no = gen_invoice_no(organization, country_code, self.invoiced_at)
         super(Invoice, self).__init__(*args, **kwargs)
 
+    @property
+    def is_final(self):
+        return self.status == INVOICE_STATUS.FINAL
+
+    @validates('invoicee_name', 'invoicee_company', 'invoicee_email', 'invoice_no', 'invoiced_at',
+    'street_address_1', 'street_address_2', 'city', 'state', 'state_code', 'country_code', 'postcode',
+    'buyer_taxid', 'seller_taxid', 'customer_order_id', 'organization_id')
+    def validate_immutable_final_invoice(self, key, val):
+        if self.status is INVOICE_STATUS.FINAL:
```

Comment:
Use `==` when comparing integers. The `is` operator only works for small numbers due to a quirk of CPython optimisation.